lib/Runtime/Debug/BreakpointProbe.cpp

```diff
@@ -12,6 +12,7 @@ namespace Js
         characterOffset(statement.statement.begin),
         byteOffset(statement.bytecodeSpan.begin)
     {
+        this->breakPointId = functionBody->GetScriptContext()->GetThreadContext()->GetDebugManager()->GetNextBreakpointId();
```

Comment:
I like it that it is controlled where BreakpointProbe is created, doesn't look like it is responsibility of the caller/user of this class.

---
In reply to: [62774317](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62774317) [](ancestors = 62774317)